app/src/main/java/emu/skyline/AppDialog.kt

```diff
@@ -117,4 +169,108 @@ class AppDialog : BottomSheetDialogFragment() {
             }
         }
     }
+
+    /**
+     * Zips the save file located in the given folder path and creates a new zip file with the name and version of the game, and the current date and time.
+     * @param saveFolderPath The path to the folder containing the save file to zip.
+     * @return true if the zip file is successfully created, false otherwise.
+     */
+    private fun zipSave(saveFolderPath : String) : Boolean {
+        try {
+            val saveFolder = File(saveFolderPath)
+            val outputZipFile = File("$savesFolderRoot${item.title} (v${binding.gameVersion.text}) [${item.titleId}] - ${LocalDateTime.now().format(DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm"))}.zip")
+            lastZipCreated?.delete()
+            outputZipFile.createNewFile()
+            ZipOutputStream(BufferedOutputStream(FileOutputStream(outputZipFile))).use { zos ->
+                saveFolder.walkTopDown().forEach { file ->
+                    val zipFileName = file.absolutePath.removePrefix(savesFolderRoot).removePrefix("/")
+                    if (zipFileName == "")
+                        return@forEach
+                    val entry = ZipEntry("$zipFileName${(if (file.isDirectory) "/" else "")}")
+                    zos.putNextEntry(entry)
+                    if (file.isFile) {
+                        file.inputStream().use { fis -> fis.copyTo(zos) }
+                    }
+                }
+            }
+            lastZipCreated = outputZipFile
+        } catch (e : Exception) {
+            return false
+        }
+        return true
+    }
+
+    private val startForResultExportSave = registerForActivityResult(ActivityResultContracts.StartActivityForResult()) { _ ->
+        lastZipCreated?.delete()
+    }
+
+    /**
+     * Exports the save file located in the given folder path by creating a zip file and sharing it via intent.
+     * @param saveFolderPath The path to the folder containing the save file(s) to export.
+     */
+    private fun exportSave(saveFolderPath : String) {
+        CoroutineScope(Dispatchers.IO).launch {
+            if (!zipSave(saveFolderPath)) {
+                withContext(Dispatchers.Main) {
+                    Snackbar.make(binding.root, R.string.error, Snackbar.LENGTH_LONG).show()
+                }
+                return@launch
+            }
+
+            withContext(Dispatchers.Main) {
+                val file = DocumentFile.fromSingleUri(requireContext(), DocumentsContract.buildDocumentUri(DocumentsProvider.AUTHORITY, "${DocumentsProvider.ROOT_ID}/switch/nand/user/save/0000000000000000/00000000000000000000000000000001/${lastZipCreated!!.name}"))!!
```

Comment:
`lastZipCreated!!` doesn't look too good. Introduce a local variable of type `File` (non-nullable), where you store a temporary copy of `lastZipCreated`. You assign it above if `zipSave` returns `true`.

See the above suggestion, use `lastZipFile` in here instead of `lastZipCreated`.